source/extensions/transport_sockets/tls/context_config_impl.cc

```diff
@@ -19,6 +21,30 @@ namespace Tls {
 
 namespace {
 
+absl::once_flag fipsOnce;
+bool fipsEnabled;
+
+void initFipsEnabled() {
+  std::ifstream file("/proc/sys/crypto/fips_enabled");
+  if (file.fail()) {
+    std::cout << "FIPS " << (fipsEnabled ? "enabled" : "disabled") << "\n";
+    fipsEnabled = false;
+  } else {
+    std::stringstream file_string;
+    file_string << file.rdbuf();
+
+    std::string fipsEnabledText = file_string.str();
+    fipsEnabledText.erase(fipsEnabledText.find_last_not_of("\n") + 1);
+    fipsEnabled = fipsEnabledText.compare("1") == 0;
+    std::cout << "FIPS " << (fipsEnabled ? "enabled" : "disabled") << ": " << fipsEnabledText << "\n";
+  }
+}
+
+bool isFipsEnabled() {
+  absl::call_once(fipsOnce, initFipsEnabled);
```

Comment:
Tricky :) In the old good days we would just use a `static` variable that would store whether or not we need to run something :)

Reply:
It seemed safer than playing directly with mutexes :)

Reply:
and statics don't work in certain memory models and regularly break